src/main/java/ch/njol/skript/expressions/ExprItemAmount.java

```diff
@@ -27,73 +27,75 @@
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.expressions.base.SimplePropertyExpression;
 import ch.njol.skript.util.slot.Slot;
-import ch.njol.util.coll.CollectionUtils;
 import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
 import org.eclipse.jdt.annotation.Nullable;
 
 @Name("Item Amount")
 @Description("The amount of an <a href='classes.html#itemstack'>item stack</a>.")
 @Examples("send \"You have got %item amount of player's tool% %player's tool% in your hand!\" to player")
-@Since("2.2-dev24")
+@Since("2.2-dev24, 2.7 (itemstacks)")
 public class ExprItemAmount extends SimplePropertyExpression<Object, Long> {
-	
-    static {
-        register(ExprItemAmount.class, Long.class, "item[[ ]stack] (amount|size|number)", "slots/itemtypes");
-    }
 
+	static {
+		register(ExprItemAmount.class, Long.class, "item[[ ]stack] (amount|size|number)", "slots/itemtypes/itemstacks");
+	}
 	
 	@Override
 	public Long convert(final Object item) {
-    	return (long) (item instanceof ItemType ? ((ItemType) item).getAmount() : ((Slot) item).getAmount());
+		if (item instanceof ItemType) {
+			return (long) ((ItemType) item).getAmount();
+		} else if (item instanceof Slot) {
+			return (long) ((Slot) item).getAmount();
+		} else {
+			return (long) ((ItemStack) item).getAmount();
+		}
 	}
-	
+
 	@Override
-    public @Nullable Class<?>[] acceptChange(Changer.ChangeMode mode) {
-        return (mode != ChangeMode.REMOVE_ALL) ? CollectionUtils.array(Number.class) : null;
-    }
+	public @Nullable Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		if (mode != ChangeMode.REMOVE_ALL)
+			return new Class[]{Number.class};
+		return new Class[0];
+	}
 
-    @Override
-    public void change(Event event, @Nullable Object[] delta, Changer.ChangeMode mode) {
-    	int amount = delta != null ? ((Number) delta[0]).intValue() : 0;
-        switch (mode) {
-            case ADD:
-            	for (Object obj : getExpr().getArray(event))
+	@Override
+	public void change(Event event, @Nullable Object[] delta, Changer.ChangeMode mode) {
+		int amount = delta != null ? ((Number) delta[0]).intValue() : 0;
+		switch (mode) {
+			case REMOVE:
+				amount = -amount;
+				// fall through
+			case ADD:
+				for (Object obj : getExpr().getArray(event))
 					if (obj instanceof ItemType) {
 						ItemType item = ((ItemType) obj);
 						item.setAmount(item.getAmount() + amount);
-					} else {
+					} else if (obj instanceof Slot) {
 						Slot slot = ((Slot) obj);
 						slot.setAmount(slot.getAmount() + amount);
+					} else {
+						ItemStack item = ((ItemStack) obj);
+						item.setAmount(item.getAmount() + amount);
 					}
-                break;
-            case SET:
+				break;
+			case REMOVE_ALL:
+			case RESET:
+			case DELETE:
```

Comment:
If we're blocking `REMOVE_ALL` in the accept change, how come it's included here?

Reply:
who tf knows, i love skript

I'll remove it, ig